packages/report-server/src/reportBuilder/transform/transform.ts

```diff
@@ -3,70 +3,78 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { yup } from '@tupaia/utils';
+
 import { Row } from '../types';
 import { transformBuilders } from './functions';
 import { aliases } from './aliases';
 
 type TransformParams = {
+  title?: string;
   name: string;
   apply: (rows: Row[]) => Row[];
 };
 
+const transformParamsValidator = yup.lazy((value: unknown) => {
+  if (typeof value === 'string') {
+    return yup
+      .mixed<keyof typeof aliases>()
+      .oneOf(Object.keys(aliases) as (keyof typeof aliases)[])
+      .required();
+  }
+
+  return yup.object().shape({
+    transform: yup
+      .mixed<keyof typeof transformBuilders>()
+      .oneOf(Object.keys(transformBuilders) as (keyof typeof transformBuilders)[])
+      .required(),
+    title: yup.string(),
+    description: yup.string(),
+  });
+});
+
+const paramsValidator = yup.array().required();
+
 const transform = (rows: Row[], transformSteps: TransformParams[]): Row[] => {
   let transformedRows: Row[] = rows;
-  transformSteps.forEach((transformStep: TransformParams) => {
-    transformedRows = transformStep.apply(transformedRows);
+  transformSteps.forEach((transformStep: TransformParams, index: number) => {
+    try {
+      transformedRows = transformStep.apply(transformedRows);
+    } catch (e) {
+      const titlePart = transformStep.title ? ` (${transformStep.title})` : '';
+      const errorMessagePrefix = `Error in transform[${index + 1}]${titlePart}: `;
+      throw new Error(`${errorMessagePrefix}${(e as Error).message}`);
+    }
```

Comment:
@edmofro hope you're okay with me sneaking this in! It's just a bit of logic to improve the error messaging when the report fails. Now you get something like:
`Error in transform[2] (Calculate occuptions from data elements): Unexpected operator ) (char 52)`

Reply:
I went with 1 based index again to be consistent... but might be a tad surprising to devs...

Reply:
Great addition!